projects/dsp-ui/src/lib/viewer/values/date-value/date-input-text/date-input-text.component.ts

```diff
@@ -0,0 +1,305 @@
+        this._disabled ? this.form.disable() : this.form.enable();
+        this.stateChanges.next();
+    }
+
+    private _disabled = false;
+
+    @Input()
+    get placeholder() {
+        return this._placeholder;
+    }
+
+    set placeholder(plh) {
+        this._placeholder = plh;
+        this.stateChanges.next();
+    }
+
+    private _placeholder: string;
+
+    @Input()
+    get required() {
+        return this._required;
+    }
+
+    set required(req) {
+        this._required = coerceBooleanProperty(req);
+        this.stateChanges.next();
+    }
+
+    private _required = false;
+
+    @HostBinding('class.floating')
+    get shouldLabelFloat() {
+        return this.focused || !this.empty;
+    }
+
+    @HostBinding() id = `dsp-date-input-text-${DateInputTextComponent.nextId++}`;
+
+    constructor(fb: FormBuilder,
+                @Optional() @Self() public ngControl: NgControl,
+                private _fm: FocusMonitor,
+                private _elRef: ElementRef<HTMLElement>,
+                @Optional() _parentForm: NgForm,
+                @Optional() _parentFormGroup: FormGroupDirective,
+                _defaultErrorStateMatcher: ErrorStateMatcher,
+                private _valueService: ValueService) {
+        super(_defaultErrorStateMatcher, _parentForm, _parentFormGroup, ngControl);
+
+        if (this.ngControl != null) {
+            // Setting the value accessor directly (instead of using
+            // the providers) to avoid running into a circular import.
+            this.ngControl.valueAccessor = this;
+        }
+
+        this.isPeriodControl = new FormControl(false); // TODO: if period, check if start is before end
+        this.calendarControl = new FormControl(null);
+
+        this.endDate = new FormControl(null);
+        this.startDate = new FormControl(null);
+
+        const eraChangesSubscription = this.isPeriodControl.valueChanges.subscribe(
+            isPeriod => {
+                this.endDate.clearValidators();
+
+                if (isPeriod && this.valueRequiredValidator) {
+                    // end date is required in case of a period
+                    this.endDate.setValidators([Validators.required]);
+                }
+
+                this.endDate.updateValueAndValidity();
+            }
+        );
+
+        this._subscriptions.push(eraChangesSubscription);
+
+        // TODO: find better way to detect changes
+        const startValueSubscription = this.startDate.valueChanges.subscribe(
+            data => {
+                // form's validity has not been updated yet,
+                // trigger update
+                this.form.updateValueAndValidity();
+                this._handleInput();
+            }
+        );
+
+        this._subscriptions.push(startValueSubscription);
+
+        // TODO: find better way to detect changes
+        const endValueSubscription = this.endDate.valueChanges.subscribe(
+            data => {
```

Comment:
is `data` needed here?

Reply:
done in https://github.com/dasch-swiss/dsp-ui-lib/pull/298/commits/1f5f6994a8de4bed0af1d580505f1ecb884dd1a7